tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1277,10 +1289,20 @@ static void SetDefaultConfigArguments(ClientContext &context) {
 	auto &config = ClientConfig::GetConfig(context);
 	config.enable_progress_bar = true;
 
-	if (DuckDBPyConnection::IsJupyter()) {
-		// Set the function used to create the display for the progress bar
-		context.config.display_create_func = JupyterProgressBarDisplay::Create;
+	if (!DuckDBPyConnection::IsJupyter()) {
```

Comment:
Just wanted to say it seems like `DuckDBPyConnection` guarantees that `ipython` is present when `::IsJupyter()` is true.
https://github.com/duckdb/duckdb/blob/v0.7.1/tools/pythonpkg/src/pyconnection.cpp#L61-L67

I think you may only need to check for `ipywidgets`. It depends on `ipython` so they should both be installed under normal circumstances anyways. Also might help making the wording in the error message less confusing.